crates/walrus-sdk/src/client.rs

```diff
@@ -441,6 +510,172 @@ impl<T: ReadClient> Client<T> {
             result = future => result,
         }
     }
+
+    /// Retrieves slivers with retry logic, only requesting missing slivers in subsequent attempts.
+    ///
+    /// This function will keep retrying until all requested slivers are received or the maximum
+    /// number of retries is reached or the timeout is reached.
+    async fn retrieve_slivers_with_retry<E: EncodingAxis>(
+        &self,
+        metadata: &VerifiedBlobMetadataWithId,
+        sliver_indices: &[SliverIndex],
+        certified_epoch: Epoch,
+        max_attempts: Option<usize>,
+        timeout_duration: Option<Duration>,
+    ) -> Result<Vec<SliverData<E>>, ClientError>
+    where
+        SliverData<E>: TryFrom<walrus_core::Sliver>,
+    {
+        let mut sliver_selector = SliverSelector::<E>::new(
+            sliver_indices.to_vec(),
```

Comment:
Nit: If you change the `SliverSelector::new` function, the `to_vec` should not be necessary.